src/common/src/telemetry/report.rs

```diff
@@ -112,3 +126,99 @@ where
     });
     (join_handle, shutdown_tx)
 }
+
+pub fn report_event_common(
+    event_stage: PbTelemetryEventStage,
+    feature_name: String,
```

Comment:
Shall we name it "event_name"? The examples in this PR are `normal_recovery`, `adhoc_recovery`, `failure_recovery` etc. which doesn't seem to be "feature" to me.

Besides, `&str` seems better, because it implies the event name must be a static string, and dynamic values should be put in `attributes`.

```suggestion
    event_name: &str,
```